components/quick_switch_modal/quick_switch_modal.tsx

```diff
@@ -156,23 +172,23 @@ export default class QuickSwitchModal extends React.PureComponent {
         }
     };
 
-    handleOnClick = (e) => {
+    private handleOnClick = (e: React.MouseEvent<HTMLAnchorElement, MouseEvent>): void => {
         e.preventDefault();
         const mode = e.currentTarget.getAttribute('data-mode');
         this.enableChannelProvider();
         this.setState({mode});
         this.focusTextbox();
     }
 
-    handleSuggestionsReceived = (suggestions) => {
-        const loadingPropPresent = suggestions.items.some((item) => item.loading);
+    private handleSuggestionsReceived = (suggestions: ProviderSuggestions): void => {
+        const loadingPropPresent = suggestions.items.some((item: any) => item.loading);
```

Comment:
Can we add a type for the suggestion items?

Reply:
I'm afraid no. items is coming from either SwitchChannelProvider or SwitchTeamProvider, both of their item arrays contain specific data set to that provider. Also both providers aren't migrated to ts, so there's no types yet.

Reply:
Theoretically, we could use an interface or a parameterized type here to wrap both types of results, but I don't think that's necessary since the providers haven't been migrated like you said. I'm fine to leave as any for now since I imagine we'll have a bunch of followup tickets to clean up anys after this whole migration is eventually done.